db/db.go

```diff
@@ -9,25 +9,42 @@ import (
 	"io/ioutil"
 	"testing"
 
+	"github.com/google/uuid"
+	"github.com/mattn/go-sqlite3"
 	"github.com/stretchr/testify/require"
 )
 
+const DealsDBName = "boost.db"
+const LogsDBName = "boost.logs.db"
+
 var ErrNotFound = errors.New("not found")
 
 type Scannable interface {
 	Scan(dest ...interface{}) error
 }
 
-func SqlDB(dbPath string) (*sql.DB, error) {
-	db, err := sql.Open("sqlite3", "file:"+dbPath)
+func SqlDB(dbPath, driverName string) (*sql.DB, *sqlite3.SQLiteConn, error) {
+	var sqlite3conn *sqlite3.SQLiteConn
+	sql.Register(driverName, &sqlite3.SQLiteDriver{
+		ConnectHook: func(conn *sqlite3.SQLiteConn) error {
+			sqlite3conn = conn
+			return nil
+		},
+	})
+	db, err := sql.Open(driverName, "file:"+dbPath)
```

Comment:
I think it may be easier to add a separate method to get the `*sqlite3.SQLiteConn` connection on demand, rather than adding it as a third return value on the existing SqlDB method

Reply:
I couldn't figure out a safe way to do this. We could try https://pkg.go.dev/database/sql#Conn.Raw but It doesn't allow driverConn must not be used outside of f. On the other hand, backup() method works on destination not source and requires a source sqliteConn{} to work.

Reply:
With this change all calls to `SqlDB()` need to pass a driverName, that is only used to create a `SQLiteConn` that most of them don't use. In each case the new driverName is added to go's sql driver registry, which is a bit strange.

I'd suggest that we
- leave the `SqlDB()` function as it is
- move sqlBackup to the db package
- create a new function in the db package to get the SQLiteConn for source and destination db
- call the new function from sqlBackup

Reply:
Redesign works as expected. I was able to take a backup and restore.